File: solutions/lua/lasagna/1/lasagna.lua
local lasagna = {
  oven_time = 40,
  prep_minutes_per_layer = 2
}

-- returns the remaining minutes based on the actual minutes in the oven
function lasagna.remaining_oven_time(actual_minutes_in_oven)
  return lasagna.oven_time - actual_minutes_in_oven
end


function lasagna.preparation_time(number_of_layers)
  return number_of_layers * lasagna.prep_minutes_per_layer
end

function lasagna.elapsed_time(number_of_layers, actual_minutes_in_oven)
  return lasagna.preparation_time(number_of_layers) + actual_minutes_in_oven
end

return lasagna

File: solutions/cpp/lasagna/1/lasagna.cpp
int ovenTime() {
    return 40;
}

int remainingOvenTime(int actualMinutesInOven) {
    return ovenTime() - actualMinutesInOven;
}

int preparationTime(int numberOfLayers) {
    int prepTimePerLayer{2};
    return numberOfLayers * prepTimePerLayer;
}

int elapsedTime(int numberOfLayers, int actualMinutesInOven) {
    return preparationTime(numberOfLayers) + actualMinutesInOven;
}
